src/saveload/saveload.cpp

```diff
@@ -2754,22 +2754,21 @@ void SetSaveLoadError(StringID str)
 	_sl.error_str = str;
 }
 
-/** Get the string representation of the error message */
-const char *GetSaveLoadErrorString()
-{
-	SetDParam(0, _sl.error_str);
-	SetDParamStr(1, _sl.extra_msg);
+/** Return the appropriate initial string for an error depending on whether we are saving or loading. */
+StringID GetSaveLoadErrorString() {
```

Comment:
Given that this and `GetSaveLoadErrorMessage()` are both dealing with StringIDs, I wonder if this might be a better name.

```suggestion
StringID GetSaveLoadErrorType() {
```

Reply:
Sure.